src/data/bucket/pattern_bucket_features.js

```diff
@@ -41,9 +41,12 @@ export function addPatternDependencies(type: string, layers: PatternStyleLayers,
 
         const patternPropertyValue = patternProperty.value;
         if (patternPropertyValue.kind !== "constant") {
-            const min = patternPropertyValue.evaluate({zoom: zoom - 1}, patternFeature, {}, options.availableImages);
-            const mid = patternPropertyValue.evaluate({zoom}, patternFeature, {}, options.availableImages);
-            const max = patternPropertyValue.evaluate({zoom: zoom + 1}, patternFeature, {}, options.availableImages);
+            let min = patternPropertyValue.evaluate({zoom: zoom - 1}, patternFeature, {}, options.availableImages);
+            let mid = patternPropertyValue.evaluate({zoom}, patternFeature, {}, options.availableImages);
+            let max = patternPropertyValue.evaluate({zoom: zoom + 1}, patternFeature, {}, options.availableImages);
+            min = min && min.name ? min.name : min;
+            mid = mid && mid.name ? mid.name : mid;
+            max = max && max.name ? max.name : max;
```

Comment:
Wasn't `null` a possible result here even before this change?

I can't remember exactly why I changed this now off the top of my head, but I think the ternary operator might have been more of a defensive guard. I'm not actually sure that `min` can ever be `null` at this point because any value supplied to `*-pattern` properties will be coerced to a ResolvedImage type. If you tried to just do `line-pattern: null` or something, you'd get a type error before it got to this point. But maybe there's a pathway I'm not thinking about.